cmd/snap/cmd_run.go

```diff
@@ -760,6 +760,29 @@ func snapdHelperPath(toolName string) (string, error) {
 	return filepath.Join(snapBase, dirs.CoreLibExecDir, toolName), nil
 }
 
+/* Kerberos tickets live in /tmp, or in the place specified by KRB5CCNAME
+ * environment variable. However, in snaps /tmp is private.
+ * So rewire the environment variable to point to /var/lib/snapd/hostfs/tmp,
+ * in case the variable is unset or resolves tickets to /tmp/krb5cc*.
+ *
+ * Snaps that want to read Kerberos tickets must then connect to the
+ * kerberos-tickets interface to have access to the tickets.
+ */
+func exposeKerberosTickets(info *snap.Info) (string, error) {
+	krb5EnvVar := osGetenv("KRB5CCNAME")
+	if len(krb5EnvVar) == 0 {
+		return "", nil
+	}
+	if strings.HasPrefix(krb5EnvVar, "FILE:") {
+		path := strings.TrimPrefix(krb5EnvVar, "FILE:")
+		path = filepath.Clean(path)
+		if filepath.Dir(path) == "/tmp" && strings.HasPrefix(path, "/tmp/krb5cc") {
+			return "FILE:/var/lib/snapd/hostfs/" + path, nil
```

Comment:
this should fix duplicate `/`:
```suggestion
			// with vanilla config this ends up returning:
			// FILE:/var/lib/snapd/hostfs/tmp/krb5cc_test
			return "FILE:" + filepath.Join("/var/lib/snapd/hostfs/", path), nil
```